content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesAction.java

```diff
@@ -10384,5 +10412,362 @@ public void doViewTrash(RunData data)
+
+			response.setContentType("application/zip;charset=UTF-8");
+			response.setHeader("Content-Disposition", "attachment; filename = site_download.zip");
+			zipOut = new ZipOutputStream(response.getOutputStream());
+
+			Iterator it = zipDownloadItems.iterator();
+			while(it.hasNext())
+			{
+				ListItem myElement = (ListItem) it.next();
+				String resourceId = myElement.getId();
+				boolean get = ContentHostingService.allowGetResource(resourceId);
+				if (get) compressResource(zipOut, collectionId, rootFolderName, resourceId);
+			}
+		}
+		catch (PermissionException pe)
+		{
+			try
+			{
+				response.sendError(response.SC_FORBIDDEN);
+			}
+			catch (IOException e)
+			{
+				e.printStackTrace();
+			}
+		}
+		catch (Throwable ignore)
+		{
+			try
+			{
+				response.sendError(response.SC_NO_CONTENT);
+			}
+			catch (IOException e)
+			{
+				e.printStackTrace();
+			}
+		}
+		finally
+		{
+			if (zipOut != null)
+			{
+				try
+				{
+					zipOut.flush();
+					zipOut.close();
+				}
+				catch (Throwable ignore)
+				{
+					ignore.printStackTrace();
+				}
+			}
+		}
+	}
+
+	protected void compressResource(ZipOutputStream zipOut, String collectionId, String rootFolderName, String resourceId) throws Exception
+	{
+		if (ContentHostingService.isCollection(resourceId))
+		{
+			try
+			{
+				ContentCollection collection = ContentHostingService.getCollection(resourceId);
+				List<String> children = collection.getMembers();
+				if(children != null)
+				{
+					for(int i = children.size() - 1; i >= 0; i--)
+					{
+						String child = children.get(i);
+						compressResource(zipOut,collectionId,rootFolderName,child);
+					}
+				}
+			}
+			catch (PermissionException e)
+			{
+				//Ignore
+			}
+		}
+		else
+		{
+			try
+			{
+				ContentResource resource = ContentHostingService.getResource(resourceId);
+				ResourceProperties props = resource.getProperties();
+				String displayName = props.getPropertyFormatted(props.getNamePropDisplayName());
+				displayName = escapeInvalidCharsEntry(displayName);
+
+				InputStream content = resource.streamContent();
+				byte data[] = new byte[1024 * 10];
+				BufferedInputStream bContent = null;
+
+				try
+				{
+					bContent = new BufferedInputStream(content, data.length);
+					
+					String entryName = (resource.getContainingCollection().getId() + displayName);
+					entryName=entryName.replace(collectionId,rootFolderName+"/");
+					entryName = escapeInvalidCharsEntry(entryName);
+
+					ZipEntry resourceEntry = new ZipEntry(entryName);
+					zipOut.putNextEntry(resourceEntry); //A duplicate entry throw ZipException here.
+					int bCount = -1;
+					while ((bCount = bContent.read(data, 0, data.length)) != -1)
+					{
+						zipOut.write(data, 0, bCount);
+					}
+
+					try
+					{
+						zipOut.closeEntry();
+					}
+					catch (IOException ioException)
+					{
+						ioException.printStackTrace();
```

Comment:
Production code should never use `e.printStackTrace()` as it can end up going to a different location to the logs and doesn't include all the useful features the loggers give (timestamps, extra context, etc). Could you switch this to use the logging framework please.